src/pip/_internal/operations/freeze.py

```diff
@@ -78,6 +80,20 @@ def freeze(
             continue
         if exclude_editable and req.editable:
             continue
+        if hashes_alg and req.canonical_name not in skip:
+            if dist.has_metadata('HASH'):
+                dist_hashes = dist.get_metadata_lines('HASH')
+                dist_selected_hash = [
+                    single_hash for single_hash in dist_hashes if (
+                        hashes_alg == single_hash.split(':')[0])]
+                if dist_selected_hash:
+                    req.hash = dist_selected_hash[0]
+            else:
+                raise FreezeError(
```

Comment:
I would probably add a logger.warn instead of raising an exception, since not being able to locate hash of a package shouldn't stop us processing other distributions

Reply:
In the original code, it was `logger.warn` :smile:
The reason I changed it in another commit, is that `requirements.txt` file with only some with `--hash` is invalid. There is no point of doing hashes at all if some distributions are missing hashes.

Reply:
If they are using the hash option to populate their requirements file, that might be valid, but if they just want to view them, we would want to let them see the hashes for distributions we can find for, and not show anything at all. But I don't feel strongly towards  this, and I am fine either way.